spectroscopy/code_src/keck_functions.py

```diff
@@ -64,7 +64,10 @@ def KeckDEIMOS_get_spec(sample_table, search_radius_arcsec):
         # Prepare arrays
         wave = spec["LAMBDA"][0] * u.angstrom
         flux_cgs = spec["FLUX"][0] * 1e-17 * u.erg/u.second/u.centimeter**2/u.angstrom
-        error_cgs = np.sqrt(1 / spec["IVAR"][0]) * 1e-17 * u.erg/u.second/u.centimeter**2/u.angstrom
+        # Inverse variances may be zero, resulting in infinite error.
+        # We'll leave these in and ignore the "divide by zero" warning.
+        with np.errstate(divide='ignore'):
+            error_cgs = np.sqrt(1 / spec["IVAR"][0]) * flux_cgs.unit
```

Comment:
This maybe a good example for a docs of good practices, e.g. deal with expected Warnings that are due to the nature of the functionality (e.g. either this or the occasional no result warning coming from astroquery when we loop through a list of coordinates where we expect to have no matches!); and any warnings the end user should/can do nothing about.